components/signoffs/signoffController.js

```diff
@@ -29,22 +29,22 @@ const addSignoff = async (req, res, next) => {
       req.body.deliverableType
     );
 
+    let createdSignoff;
     const validationResults = validateObjectsFromUser(req.body);
     if (existingSignoffs == null || !existingSignoffs.length) {
-      const createdSignoff = await upsert(
+      let createdSignoff = await upsert(
         req.params.caseIdentifier,
         validationResults
       );
-      return res.status(201).json(createdSignoff);
     } else {
       // signoff step + deliverable are same, delete the old signoff record and add new one
-      const createdSignoff = await upsert(
+      let createdSignoff = await upsert(
```

Comment:
Need to remove `let` here (and on L35) as this is defining a new block-level variable and hiding outer variable, leaving the value you're returning undefined